### DISCUSSION TITLE
How to add deviance metrics with sample weights?

### DISCUSSION THREAD

**Discussion by StrudelDoodleS** [autogluon/autogluon#3945] [Q&A]
Hi everyone.

I am new to using this package and I have read the [docs](https://auto.gluon.ai/1.0.0/tutorials/tabular/advanced/tabular-custom-metric.html) for adding custom scorers.

I am specifically trying to add [mean_poisson_deviance](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.mean_poisson_deviance.html) and [mean_gamma_deviance](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.mean_gamma_deviance.html#sklearn.metrics.mean_gamma_deviance). Both of these I beleive are just mean_tweedie_deviance in disguise just using a tweedie variance power of 1 or 2 respectively.

Using this code: `make_scorer(name='mpd',
                                 score_func=sklearn.metrics.mean_poisson_deviance,
                                 optimum=0,
                                 greater_is_better=False)` results in some odd behaviour. Because `optimum=0` the sigh doesn'tmatter and we just end up with `-score` instead (this is referenced in the docs. Also parsing sample_weight to the TabularPredictor class it sas this eval metric does not take a sample_weight but it very much does.

The end result is a lot of models fail to fit as we get the y_pred must be strictly positive ValueError. I have double and triple chekced and given my target is count data nothing is vbelow zero and there are no sign errors.
                                 
I'm not sure what to about this other than maybe create a custom scorer of some kind. But I am having trouble coming up with one. If anyone has dealt with this issue before let me know.

Thanks! :)
